src/Components/WebAssembly/WebAssembly.Authentication/src/Services/DeserializedAuthenticationStateProvider.cs

```diff
@@ -11,13 +11,13 @@ namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication;
 
 internal sealed class DeserializedAuthenticationStateProvider : AuthenticationStateProvider
 {
-    // Do not change. This must match all versions of the server-side AuthenticationStateSerializer.PersistenceKey.
-    private const string PersistenceKey = $"__internal__{nameof(AuthenticationState)}";
-
     private static readonly Task<AuthenticationState> _defaultUnauthenticatedTask =
         Task.FromResult(new AuthenticationState(new ClaimsPrincipal(new ClaimsIdentity())));
 
-    private readonly Task<AuthenticationState> _authenticationStateTask = _defaultUnauthenticatedTask;
+    private readonly Task<AuthenticationState> _authenticationStateTask;
+
+    [SupplyParameterFromPersistentComponentState]
+    private AuthenticationStateData? AuthStateData { get; set; }
 
     [UnconditionalSuppressMessage(
```

Comment:
After the last change, we do not pass `PersistentComponentState` as an argument anymore. Why would we keep `DynamicDependency` declarations?